dangvuthang/rocodingtest
# Crud dashboard

### ADDED CRUD to Dashboard:

- Post, delete, get, patch has been implemented.

**Description how to test:** 

- npm run dev: frontend and backend
- run MongoCompass and Postman
- go to address: localhost:3000/dashboard
- press **sign in** button -> to get the token -> to view token write console.log(accessToken) under line 44 in dashboard.tsx

1. ### Test add exam:

- click on the add exam button in dashboard
- add information to the form and click add exam
- check status success in **console** and **MongoDB - tests**

2. ### Test edit exam:
_**(Note: the endpoint must have a similar id in the db)**_
-  change endpoint at **line 51** : `/tests/_id` ( example:  url: `/tests/625d4e2ca9d3d42737bd46b4`) -> to display the card
- change endpoint at **line 109** : `/tests/_id` ( example:  url: `/tests/625d4e2ca9d3d42737bd46b4`) -> to edit the content of the card
- then proceed to dashboard.
- click **edit exam button** on one of the card in dashboard.
- add different information to the form and click update.
- check status success in **console** and **MongoDB - tests** and postman.

3. ### Test delete exam:
_**(Note: the endpoint must have a similar id in the db)**_
-  change endpoint at **line 51** : `/tests/_id ` ( example:  url: `/tests/625d4e2ca9d3d42737bd46b4`) -> to display the card.
-  change endpoint at **line 91** : `/tests/_id` ( example:  url: `/tests/625d4e2ca9d3d42737bd46b4`) -> to delete the card 
- then proceed to dashboard 
- click on edit exam on one of the card in dashboard
- click **delete button** on the card
- check the result in **console** and **MongoDB - tests** and postman
